paramiko/transport.py

```
@@ -1746,16 +1748,16 @@ def run(self):
                         self._send_message(msg)
                     self.packetizer.complete_handshake()
             except SSHException as e:
-                self._log(ERROR, traceback.format_exc())
+                self._log(ERROR, "SSH Exception")
                 self.saved_exception = e
             except EOFError as e:
                 self._log(DEBUG, 'EOF in transport thread')
                 self.saved_exception = e
             except socket.error as e:
-                self._log(ERROR, traceback.format_exc())
+                self._log(ERROR, "Socket Error")
                 self.saved_exception = e
             except Exception as e:
-                self._log(ERROR, traceback.format_exc())
+                self._log(ERROR, "General Exception")
```

why mask the original exception traceback?

---

The _log function will propagate the last traceback created. Therefore there is no need to throw it in as a string. Plus by using the original traceback instead of the string we get better details in Sentry.

---

oh ok! good.